### PR TITLE
Some sanity checks/simplifications around key deletion

### PR DESCRIPTION
**Pull request by JayH5**

### CODE REVIEW

File: consular/main.py [hunk 1]
```diff
@@ -413,6 +419,8 @@ def get_consul_consular_keys(self):
     def delete_consul_kv_keys(self, keys, recurse=False):
         """ Delete a sequence of Consul k/v keys. """
         for key in keys:
+            log.msg('Deleting Consul k/v key "%s", recursively? %s' % (
```
Comment (miltontony) [inline consular/main.py:429]: Why is this a question?

Comment (JayH5) [inline consular/main.py:429]: It'll look like `Deleting Consul k/v key "my-app", recursively? True`

Comment (miltontony) [inline consular/main.py:429]: :) cool

File: consular/main.py [hunk 2]
```diff
@@ -457,30 +466,19 @@ def purge_dead_app_labels(self, apps):
...
+                raise e
 
         # Filter the present apps out
-        d.addCallback(self._filter_marathon_apps, apps)
+        keys = self._filter_marathon_apps(keys, apps)
 
         # Delete the remaining keys
-        return d.addCallback(self.delete_consul_kv_keys, recurse=True)
-
-    def _ignore_not_found_error(self, failure, return_value=None):
```
Comment (miltontony) [inline consular/main.py:469]: Why are you replacing this approach with the `try/catch`?

**Comment by miltontony** [comment]
:+1:

File: consular/main.py (hunk 2 above)
Comment (JayH5) [inline consular/main.py:469]: Basically, I don't trust that the callback/errback chain is working as I expect it to. It's a sanity check really.

**Pull request merged**